### PR TITLE
Build a blog page with a comments feature.

### PR DESCRIPTION
**Pull request by SydneyCaulfeild**
I have created a CommentServlet.java with a doPost() do save user comments to Datastore and a doGet() to retrieve the comments. I have updated script.js to build a UI for the comments. I also created individual css files for each page.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/add-comment")
+public class CommentServlet extends HttpServlet {
+  
+ @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    int entityCount=-1;
```
Comment (lipczak-g) [inline portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java:26]: style comment: please use spaces before and after '='

File: portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    List<Comment> commentList = new ArrayList<>();
+
+    for (Entity entity: results.asIterable()) {
+
+      String name = (String) entity.getProperty("name");
+      String message = (String) entity.getProperty("message");
+      Date time = (Date) entity.getProperty("timestamp");
+
+      commentList.add(new Comment(name, message, time));
+      entityCount++;
```
Comment (lipczak-g) [inline portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java:42]: it looks to me that commentList.size() will give you the same number without the need of a variable.

File: portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
```
Comment (lipczak-g) [inline portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java:44]: This looks much cleaner to me:

int numCommentsRequested = getNumComments(request);
if (commentList.size() > numCommentsRequested) {
  commentList = commentList.subList(0, numCommentsRequested);
}

File: portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  private int getNumComments(HttpServletRequest request){
+    String userChoiceString = request.getParameter("commentsQuantity");
+
+    int userChoice;
+    try {
+      userChoice = Integer.parseInt(userChoiceString);
+    } 
+    catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + userChoiceString);
+      return -1;
```
Comment (lipczak-g) [inline portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java:68]: don't be afraid to throw exceptions, especially when the special case int is not really helping much - I think you will get an exception in the sublist call.

In general, we try to follow https://en.wikipedia.org/wiki/Fail-fast philosophy.

File: portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    // Check that the input is between 1
+    if (userChoice < 1) {
+      System.err.println("Player choice is out of range: " + userChoiceString);
```
Comment (lipczak-g) [inline portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java:73]: try avoiding system.err prints - if you need it, exception likely makes more sense, in this case:
https://docs.oracle.com/javase/7/docs/api/java/lang/IllegalArgumentException.html

**Review by lipczak-g — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java (hunk 2 above)
Comment (SydneyCaulfeild) [reply portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java:42]: Yes this worked thanks!

**Pull request merged**